src/cli_utils.rs

```diff
@@ -75,25 +76,32 @@ pub fn print_header<W: Write>(sink: &mut W, row: &str, columns: usize) -> io::Re
     writeln!(sink, "{}", row)
 }
 
-pub fn print_results<'a, I, W>(
+pub fn print_results<'a, I, W, F>(
     sink: &mut W,
     row: &str,
     languages: I,
     list_files: bool,
+    num_format: &F,
 ) -> io::Result<()>
 where
     I: Iterator<Item = (&'a LanguageType, &'a Language)>,
     W: Write,
+    F: NumFormat,
 {
     let path_len = row.len() - NO_LANG_ROW_LEN_NO_SPACES;
     let columns = row.len();
     for (name, language) in languages.filter(isnt_empty) {
-        print_language(sink, columns, language, name.name())?;
+        print_language(sink, columns, language, name.name(), num_format)?;
 
         if list_files {
             writeln!(sink, "{}", row)?;
             for stat in &language.stats {
-                writeln!(sink, "{:1$}", stat, path_len)?;
+                writeln!(
+                    sink,
+                    "{:1$}",
+                    FormattableStats::new(stat, num_format),
```

Comment:
I'm not a big fan of the `FormattableStats` newtype, since it's only used for displaying the stats. I think what would an improvement would be to add a `display` method to `Stats` that takes `NumberFormatStyle` as an argument. This way the formatting API is also available to library authors.

```rust
pub fn display(&self, style: NumberFormatStyle) -> Result<String> {
}
```

I would also keep the `fmt::Display` implementation of `Stats` if possible since removing it will be a breaking change. So for now I'd prefer to keep that, and add a warning (`log::warn!`) to transition people to using the new method.

Reply:
I didn't think about the breaking change! That's a good point.

The only hard part about implementing `display` as you've suggested is that we don't have `fmt::Formatter`, which means we don't have a width. Is that a blocker?

Reply:
You can add other arguments, what I said was a suggestion.